packages/igx-templates/igx-ts/custom-templates/crm-grid/files/src/app/__path__/__filePrefix__.component.ts

```diff
@@ -34,169 +36,205 @@ import { data } from './data';
+            this.getDeals(employee);
+        }
+        this.localData = employees;
+    }
+
+    public toggleHiding() {
+        this._overlaySettings.positionStrategy.settings.target = this.hidingButton.nativeElement;
+        this.toggleRefHiding.toggle(this._overlaySettings);
+    }
+
+    public togglePinning() {
+        this._overlaySettings.positionStrategy.settings.target = this.pinningButton.nativeElement;
+        this.toggleRefPinning.toggle(this._overlaySettings);
+    }
+
+    public ngAfterViewInit() {
+        this.cols = this.grid1.columnList;
+        this.hiddenColsLength = this.cols.filter((col) => col.hidden).length;
+        this.pinnedColsLength = this.cols.filter((col) => col.pinned).length;
+        this.grid1.toolbar.columnPinningDropdown.width = "250px";
+
+        this.grid1.toolbar.excelExporter.onColumnExport.subscribe((args: IColumnExportingEventArgs) => {
+            if (args.field === "Deals") {
+                args.cancel = true;
+            }
+        });
+
+        this.grid1.toolbar.csvExporter.onColumnExport.subscribe((args: IColumnExportingEventArgs) => {
+            if (args.field === "Deals") {
+                args.cancel = true;
+            }
+        });
+    }
+
+    public toggleVisibility(col: IgxColumnComponent) {
+        if (col.hidden) {
+            this.hiddenColsLength--;
+        } else {
+            this.hiddenColsLength++;
+        }
+        col.hidden = !col.hidden;
+    }
+
+    public togglePin(col: IgxColumnComponent, evt) {
+        if (col.pinned) {
+            this.grid1.unpinColumn(col.field);
+            this.pinnedColsLength--;
+        } else {
+            if (this.grid1.pinColumn(col.field)) {
+                this.pinnedColsLength++;
+            } else {
+                // if pinning fails uncheck the checkbox
+                evt.checkbox.checked = false;
+            }
+        }
+    }
+
+    public formatDate(val: Date) {
+        if (!this.frmt) {
+            this.frmt = new Intl.DateTimeFormat("en-US");
+        }
+        return this.frmt.format(val);
+    }
+
+    public searchKeyDown(ev) {
+        if (ev.key === "Enter" || ev.key === "ArrowDown" || ev.key === "ArrowRight") {
+            ev.preventDefault();
+            this.grid1.findNext(this.searchText, this.caseSensitive);
+        } else if (ev.key === "ArrowUp" || ev.key === "ArrowLeft") {
+            ev.preventDefault();
+            this.grid1.findPrev(this.searchText, this.caseSensitive);
+        }
+    }
+
+    public updateSearch() {
+        this.caseSensitive = !this.caseSensitive;
+        this.grid1.findNext(this.searchText, this.caseSensitive);
+    }
+
+    public clearSearch() {
+        this.searchText = "";
+        this.grid1.clearSearch();
+    }
+
+    public formatValue(val: any): string {
+        return val.toLocaleString("en-us", { maximumFractionDigits: 2 });
+    }
+
+    public getDeals(employee: any): any {
+        employee["Deals"] = this.getDealsData();
```

Comment:
Assignment should not be done via string literal to avoid lint errors:
```suggestion
        employee.Deals = this.getDealsData();
```